### ISSUE TITLE
Reorganize the hierarchy of the CLI to be more action oriented

### ISSUE THREAD

**Issue by russorat** [influxdata/influxdb#17655]
```
$ influx get buckets
$ influx export
$ influx apply pkg
$ influx config ...
```

instead of having the rest hierarchy we have now, we orient it around the action users want to take. I think it supports what you are after more so than cobbling it together in its current shape.

@jsteenb2 can add additional details here.

**Comment by mark-rushakoff** [comment]
I agree that `get buckets` reads better, and when you know what you want to do, it writes better; but it is extremely less discoverable. 

If I run `influx help` I want to discover what objects I can operate on. Seeing that I can run `get` or `delete` or `find` is minimally helpful. Seeing that I can operate on `buckets`, `orgs`, `users` etc is actually helpful. And then running `influx buckets help` to see the operations available, will show me the exact set of operations that apply only to buckets.

If we change from `influx <object> <verb>` to `influx <verb> <object>`, how do you see the `--help` output changing?

**Comment by jsteenb2** [comment]
@mark-rushakoff I hear you. Right now we support about half the resources we have in the platform. The `-h` usage is pretty big already and only grows as we extend things.  The verb `-h` would explain the actions themselves and provide what they can do.

```
$ influx -h

get         retrieve resources ....; Example get buckets
apply     apply resources from a package...
export   export resource(s) to a package...
config    ... yada yada
...
```

don't quote me on that write up, but something to that affect. Interestingly, I find the current `-h` so busy I miss things in it. I like the separation here, filters down the root commands and nukes a bunch of filter commands that do nothing.

For instance:

```
influx buckets
```

this command does nothing except display the child commands. Every resource has this structure (minus pkg command). 

It also leads to behavioral anomalies which has affected the influx CLI. I want to see it more uniform. And eventually, when the API designs become more standard, be able to generate the entire CLI. Those are my long tail dreams for the CLI :grimacing: . This new design is a step in that direction for standardizing the entire CLI.

In short, it being `extremely less discoverable` is debatable. I'd wager its more discoverable, extensible, and uniform.

**Comment by mark-rushakoff** [comment]
I see. Maybe my mental model is stuck back in the >1y ago ways, where we were less consistent on the `verb`s used. If the list of verbs is small and fixed like `get`/`find`/`update`/`delete`, then that isn't too bad. 

I would still like a clear way to discover what resources you can operate on, to remove ambiguity around "is it `get orgs` or `get organizations`"? That seems fine if you can discover that by `influx get --help`, but there ought to be a more direct way to discover that list. In any case, that can be separate from this issue.

We may need a period of supporting the old ways while warning that it is deprecated, to give people a chance to migrate existing scripts using `influx bucket get` to `influx get bucket`.

**Comment by jsteenb2** [comment]
@mark-rushakoff absolutely agree with you. I think we'll end up with a small number of root commands.  

heads up, we have support for shell completions in existing `influx` CLI. Running `influx completion -h` will provide help for setting it up. For me using zsh, all I have to do is `source <(influx completion zsh)` and everything is gravy. I dumped that line into my dotfile so that my influx ClI completions are always up to date. Works like a charm

**Comment by jsteenb2** [comment]
@mark-rushakoff I updated the issue here with a more thorough description of what I envisioned. Would greatly appreciate your thoughts.

**Comment by rbetts** [comment]
_Discovering the resource types_: What about rendering some of the swagger resource definitions via a CLI action? I have a hard time finding the swagger rendering. Having it in the CLI might be useful to others and help around resource discovery?

_Using the term `export`_: at the risk of bike-shedding, we've use "export" to mean "backup" in 1.x [^1]. Using `backup` and `restore` seems fine to me. Just FYI.

_About beta_: We likely have cloud2 customers using the CLI? Also just FYI for transition planning. We can probably identify those users from logs if we make the CLI send an identifiable header / user-agent string?

_Consistency with `influxd` options_: I don't know if this matters, honestly. We have been trying to put all the functionality that runs against a running instance in the `influx` command and all the functionality that runs against offline data in the `influxd` command. I don't know that there needs to be any real alignment between the two tools as they are quite different; however, probably worth considering ahead of beta if reviewing CLI structure a final time.

[^1]: https://docs.influxdata.com/enterprise_influxdb/v1.7/administration/backup-and-restore/#exporting-data

**Comment by jsteenb2** [comment]
@rbetts great points. Rending swagger via a CLI action, definitely doable. If we have the open api spec, we can even generate curl commands and whatever else we want. Having a spec is incredibly powerful 😃. My long term plan/dream is to have the open api spec generate good chunks of the CLI so it is always up to date.  Of course the open api spec itself will be generated from the handlers (always up to date) in this scenario. 

This is part and parcel of the service definition work in my mind.... when we have a clear and repeatable pattern, things are much easier to maintain and extend :-)

**Comment by mhall119** [comment]
There's no reason `influx help buckets` couldn't return information showing you `influx list buckets`, `influx export buckets`, etc.
